### PR TITLE
Adds a transformation to preserve the CDF of the input data

### PR DESCRIPTION
**Pull request by j-bryan**
--------
Pull Request Description
--------
##### What issue does this change request address? (Use "#" before the issue to link it, i.e., #42.)
#2164 

##### What are the significant changes in functionality due to this change request?
- Adds an option to invert a SKLTransformer derived class, which switches the forward and inverse transformation functions
- Adds `getClusteringValues` method to SKLCharacterizer to allow SKLCharacterizer derived classes to be used for clustering
- Adds `Gaussianize` transformer
- Adds `gaussianize` setting to ARMA and changes the default value to False
- Updates and regolds several SyntheticHistory tests and the TSACharacterizer.Basic test to reflect the change in default value for `gaussianize`
- Makes a handful of minor fixes to docstrings, error message formatting, etc.

----------------
For Change Control Board: Change Request Review
----------------
The following review must be completed by an authorized member of the Change Control Board.
- [x] 1. Review all computer code.
- [x] 2. If any changes occur to the input syntax, there must be an accompanying change to the user manual and xsd schema. If the input syntax change deprecates existing input files, a conversion script needs to be added (see Conversion Scripts).
- [x] 3. Make sure the Python code and commenting standards are respected (camelBack, etc.) - See on the [wiki](https://github.com/idaholab/raven/wiki/RAVEN-Code-Standards#python) for details.
- [x] 4. Automated Tests should pass, including run_tests, pylint, manual building and xsd tests. If there are changes to Simulation.py or JobHandler.py the qsub tests must pass.
- [x] 5. If significant functionality is added, there must  be tests added to check this. Tests should cover all possible options.  Multiple short tests are preferred over one large test. If new development on the internal JobHandler parallel system is performed, a cluster test must be added setting, in <RunInfo> XML block, the node ```<internalParallel>``` to True.
- [x] 6. If the change modifies or adds a requirement or a requirement based test case, the Change Control Board's Chair or designee also needs to approve the change.  The requirements and the requirements test shall be in sync.
- [x] 7. The merge request must reference an issue.  If the issue is closed, the issue close checklist shall be done.
- [x] 8. If an analytic test is changed/added is the the analytic documentation updated/added?
- [x] 9. If any test used as a basis for documentation examples (currently found in `raven/tests/framework/user_guide` and `raven/docs/workshop`) have been changed, the associated documentation must be reviewed and assured the text matches the example.

### CODE REVIEW

**Comment by joshua-cogliati-inl** [comment]
FYI: We merged some changes to RAVEN, and now there are conflicts that need to be resolved.  Sorry.

File: doc/user_manual/generated/internalRom.tex [hunk 1]
```diff
@@ -2226,6 +2236,19 @@ \subsubsection{SyntheticHistory}
             sets a seed for the underlying random number generator, if present.
       \end{itemize}
 
+    \item \xmlNode{preservecdf}:
```
Comment (dylanjm) [inline doc/user_manual/generated/internalRom.tex:2239]: I believe this should be `preserveCDF` now

File: ravenframework/TSA/Transformers/Normalizers.py [hunk 2]
```diff
@@ -22,7 +22,7 @@
 import sklearn.preprocessing as skl
 
 from .ScikitLearnBase import SKLTransformer, SKLCharacterizer
-from ...utils import InputTypes
+from ...utils import InputTypes, mathUtils
```
Comment (dylanjm) [inline ravenframework/TSA/Transformers/Normalizers.py:25]: Why add `mathUtils` here, does it get used?

**Review by dylanjm — CHANGES_REQUESTED** [review]
Just two minor comments for consideration

File: ravenframework/TSA/Transformers/Normalizers.py (hunk 2 above)
Comment (j-bryan) [reply ravenframework/TSA/Transformers/Normalizers.py:25]: Ah, that was left over from when I had some of these distribution transformations in that file. I'll remove it.

**Review by dylanjm — APPROVED** [review]

**Review by dylanjm — APPROVED** [review]
